pkg/types/validation/installconfig.go

```diff
@@ -87,6 +87,9 @@ func ValidateInstallConfig(c *types.InstallConfig) field.ErrorList {
 		// FIX-ME: As soon bz#1915122 get resolved remove the limitation of 14 chars for the clustername
 		nameErr = validate.ClusterNameMaxLength(c.ObjectMeta.Name, 14)
 	}
+	if c.Platform.IBMCloud != nil {
+		nameErr = validate.ClusterNameIBMRestrictions(c.ObjectMeta.Name)
```

Comment:
This validation accepts cluster names that would be rejected by `validate.ClusterName`. It looks like there is actually a bug in the validation for ovirt where the validation is only checking the length and ignoring any errors from `validate.ClusterName`. Let's not make the same mistake here. My suggestion is to either (1) create a new function in the `validate` package that will accept a max length and do the same thing that `validate.ClusterName` does or (2) validate the length first and then call `validate.ClusterName` if and only if the length is valid.

Reply:
Your changes now change the behavior for other platforms. Please keep your changes specific to the ibmcloud platform.

Reply:
Ah, yes, I think I follow what you mean, given platforms can override the `error` with `nil` potentially.

Reply:
Not sure if the new approach is acceptable (since `validate.ClusterName` is still run first), as I think it reduces code complexity.